crates/mako/src/visitors/import_meta_env_replacer.rs

```diff
@@ -0,0 +1,126 @@
+use swc_core::common::DUMMY_SP;
+use swc_core::ecma::ast::{
+    Expr, Ident, KeyValueProp, MemberExpr, MemberProp, MetaPropExpr, MetaPropKind, ObjectLit, Prop,
+    PropOrSpread,
+};
+use swc_core::ecma::atoms::js_word;
+use swc_core::ecma::utils::{quote_ident, quote_str};
+use swc_core::ecma::visit::{VisitMut, VisitMutWith};
+
+#[derive(Debug)]
+pub(crate) struct ImportMetaEnvReplacer {
+    pub(crate) mode: String,
+}
+
+impl ImportMetaEnvReplacer {
+    pub(crate) fn new(mode: String) -> Self {
+        Self { mode }
+    }
+}
+
+impl VisitMut for ImportMetaEnvReplacer {
+    fn visit_mut_member_expr(&mut self, member_expr: &mut MemberExpr) {
+        match member_expr {
+            MemberExpr {
+                obj:
+                    box Expr::Member(MemberExpr {
+                        obj:
+                            box Expr::MetaProp(MetaPropExpr {
+                                kind: MetaPropKind::ImportMeta,
+                                ..
+                            }),
+                        prop:
+                            MemberProp::Ident(Ident {
+                                sym: js_word!("env"),
+                                ..
+                            }),
+                        ..
+                    }),
+                ..
+            } => {
+                // replace import.meta.env.MODE with "({ MODE: 'production' }).MODE"
+                *member_expr.obj = Expr::Paren(swc_core::ecma::ast::ParenExpr {
+                    span: DUMMY_SP,
+                    expr: ObjectLit {
+                        props: vec![PropOrSpread::Prop(
+                            Prop::KeyValue(KeyValueProp {
+                                key: quote_ident!("MODE").into(),
+                                value: quote_str!(self.mode.clone()).into(),
+                            })
+                            .into(),
+                        )],
+                        span: DUMMY_SP,
+                    }
+                    .into(),
+                });
+            }
+
+            _ => member_expr.visit_mut_children_with(self),
+        }
+    }
+
+    fn visit_mut_expr(&mut self, expr: &mut Expr) {
```

Comment:
visit_mut_expr and visit_mut_member_expr are duplicated, maybe just to keep the visist_mut_expr is enough.

but add the below case

```js
let x = () => import.meta.env
```